tests/IntegrationTests.py

```diff
@@ -10,38 +15,25 @@
 import sys
 import os
 
+from selenium.webdriver.support.wait import WebDriverWait
+
 from .utils import assert_clean_console, switch_windows
 
 TIMEOUT = 120
+from selenium.webdriver.support import expected_conditions as EC
 
 
 class IntegrationTests(unittest.TestCase):
     def wait_for_element_by_css_selector(self, selector):
-        start_time = time.time()
-        latest_exception = None
-        while time.time() < start_time + TIMEOUT:
-            try:
-                return self.driver.find_element_by_css_selector(selector)
-            except Exception as e:
-                latest_exception = e
-                pass
-            time.sleep(0.25)
-        if latest_exception:
-            raise latest_exception
+        return WebDriverWait(self.driver, 10).until(
```

Comment:
Neat, I didn't know about `WebDriverWait`. Don't you want to use `TIMEOUT` instead of `10` though?